### DISCUSSION TITLE
Another field with a photo in the collection

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#13356] [Archive]
Why is there an error when creating a second field with a photo and the relationship does not arise?
I create the first field with photo with the same settings...
Directus 9.10.0 and MS SQL

```
  "errors": [
    {
      "message": "ALTER TABLE [test] ADD CONSTRAINT [test_photo2_foreign] FOREIGN KEY ([Photo2]) REFERENCES [directus_files] ([id]) ON DELETE CASCADE - Could not create constraint or index. See previous errors.",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
```

**Comment by rijkvanzanten** [comment]
Try creating the relationship without the CASCADE delete when adding the field. You can configure that under the Relationship tab 👍🏻

**Reply by koksikus** [reply]
Hi, unfortunately I've tried everything and can't do it. 
Next I did it on the server side (I created relationships directly through SQL Management Studio and it works), and when I want to change in the field On Delete of directus_files, the same error pops up in Directus. Is it a system error or am I doing something wrong?
  "errors": [
    {
      "message": "ALTER TABLE [Test_Results] ADD CONSTRAINT [test_results_test_file_2_foreign] FOREIGN KEY ([Test_File_2]) REFERENCES [directus_files] ([id]) ON DELETE SET NULL - Could not create constraint or index. See previous errors.",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"

Latest Directus 9.10.0, Node 16.15.0, locally and MSSQL 2017.

**Reply by koksikus** [reply]
@rijkvanzanten What do you think about it? Have you checked it and it works for you in this server configuration?

**Reply by rijkvanzanten** [reply]
Just answered it on the new issue you've created: https://github.com/directus/directus/issues/13488#issuecomment-1135026411
